internal/common/utils.go

```diff
@@ -44,14 +53,52 @@ func UpdateOperatingState(name string, state string, lc logger.LoggingClient, dc
 	}
 }
 
-func SendEvent(event *dtos.Event, correlationID string, lc logger.LoggingClient, ec interfaces.EventClient) {
+func SendEvent(event *dtos.Event, correlationID string, dic *di.Container) {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	configuration := container.ConfigurationFrom(dic.Get)
 	ctx := context.WithValue(context.Background(), CorrelationHeader, correlationID)
 
-	req := requests.NewAddEventRequest(*event)
-	res, err := ec.Add(ctx, req)
-	if err != nil {
-		lc.Errorf("failed to push event to core-data: %s", err)
+	if configuration.MessageQueue.Enabled {
+		encoding := clients.ContentTypeJSON
+		mc := container.MessagingClientFrom(dic.Get)
+		for _, r := range event.Readings {
+			if r.ValueType == v2.ValueTypeBinary {
+				encoding = clients.ContentTypeCBOR
+				break
+			}
+		}
+
+		var err error
+		var bytes []byte
+		switch encoding {
```

Comment:
this switch case is similar to what we've done in EventClient:
https://github.com/edgexfoundry/go-mod-core-contracts/blob/1c0e349fb4299bf449335fb3569ee365a1b383ad/v2/clients/http/utils/common.go#L88-L99

Reply:
Yes, would be nice if we could refactor that code to a function so that it is reusable here. Then I don't see the need to try to have a `messagebus` version of Event Client as I previously suggested. I don't see it being needed elsewhere.

Reply:
Are we going to put `AddEventRequest` in the `MessageEnvelope` or just `Event`? (in this PR I put `Event`)
The former maybe we can add a helper method under `AddEventRequest` to encode the whole struct based on readings value.

Reply:
ohhh, I missed that! Needs to be `AddEventRequest` same as if if was coming from Core Data.

Reply:
ok then I'm gonna see if we can do some refactoring in the go-mod-core-contracts side.